### ISSUE TITLE
Forks of my repository are ALSO posted on NuGet

### ISSUE THREAD

**Issue by SoftCircuits** [NuGet/NuGetGallery#9180]
I have a general question about when people fork one of my repositories, and then post their fork on NuGet.

For example, one of my repositories is [Silk](https://www.nuget.org/packages/SoftCircuits.Silk/). This has several forks on GitHub. I don't understand all the details of forks. But one of them also posted their fork on [NuGet](https://www.nuget.org/packages/LbsFeldkirch.Silk/).

My question is about this practice. Are there valid reasons for someone to do this? Should there be a way that I can prevent other people from listing my repositories on NuGet? Their fork is getting downloads while I'm trying to attract more attention to my NuGet packages. In fact, it's getting more downloads than mine!

**Comment by zhhyu** [comment]
Hi, @SoftCircuits. If you have concerns about any copyright issues, please reach out to us:
https://docs.microsoft.com/en-us/nuget/nuget-org/policies/deleting-packages#exceptions

Thanks!

**Comment by agr** [comment]
This is the nature of open source projects: publishing your code publicly under open source license usually grants others the right to copy, modify and redistribute your code, so much that GitHub makes it really easy to fork public repositories.

As to why it happens in your specific case, only the person who made the fork would be able to answer, you can try checking the changes they made after forking to get the idea or you can try contacting the package owners using the "Contact owners" link on the package details page:
![image](https://user-images.githubusercontent.com/102933/184459129-e080ebb4-2fbb-4b56-b14f-10b645ed3ad5.png)

**Comment by SoftCircuits** [comment]
@agr I understand it's open source, which allows them to do things with the source code. But publishing it as your own NuGet package with--as far as I can tell--no changes, and getting more downloads than mine just seems to be a problem.

I may contact them, but I'm not sure I can do anything more than ask nicely if they'd take ut down.

**Comment by agr** [comment]
You are comparing noise to noise: when we are talking about dozens downloads per month, those are most likely generated by bots that scrape the package feed for whatever reason. Packages with absolutely zero utility generate a few downloads per month.

And regarding the changes: quick glance at the other package repository shows that at the very least, the package was retargeted to net6.0 and some `CancellationToken` support was added here and there, so I wouldn't say there are no changes.

**Comment by SoftCircuits** [comment]
@agr It already targeted .NET 6.0. The main change he made was to replace every occurrence of my name with his name.

**Comment by joelverhagen** [comment]
## My perspective on the situation

Hello @SoftCircuits, over the years these problems have come up from time to time. There's a lot of different things that could be happening here:

1. As @agr, the fork may introduce legitimate changes or enhancements specific to the editor's needs. 
   - Based on the description of the package you provided and looking at the code, you project indeed _supports_ .NET 6 (with a .NET 5 and .NET Standard https://github.com/SoftCircuits/Silk/blob/7cc8a9e8bbf0dfadfaf0401147b60aaf93c22119/Silk/Silk.csproj#L4) but the fork directly targets .NET 6 (https://github.com/KurtLinder/Silk/blob/66d47aed1fc307d21efa6e15cee7eb27f0658089/Silk/Silk.csproj#L4). I can't guess why the fork has this but perhaps the fork author was encountering some problem around compatibility.
1. Sometimes the original source code is not released to NuGet.org yet and the fork author wanted to unblock themselves so they pushed a build to NuGet.org.
   - This used to happen much more as NuGet was still gaining popularity in the .NET ecosystem.
1. Sometimes we (as NuGet.org owners) never get to the bottom of why the fork exists. Perhaps the fork owner is unresponsive to the original author's messages or attempts at contact.
1. Sometimes the situation does end up being negative and the fork is indeed trying to maliciously copy the source code.
    - These situations are very rare partly because it's hard to conclusively determine this. Malicious code in packages is taken very seriously by my team but is still a relatively rare occurrence.

For the first, second, and third situation we (meaning you the original author and us the NuGet.org team) should try to assume positive intent and then take next steps to improve the situation as much as possible. As an aside, in my heart, I really believe most developers want to do the right thing but maybe don't have all of the tools or knowledge to fork/copy a package in a manner that is fully agreeable to the author. One original author might have different feelings than another about what the right way to fork a package is. It's honestly a murky area and the OSS licensing terms for your code are just the starting point for a conversation like this.

The fact that the fork actually mentioned the original source and left your copyright string as-is suggests to me that the fork author was trying their best leave a trace of where they got the code from and provide credit where credit is due.
![image](https://user-images.githubusercontent.com/94054/186694846-671166cc-5b11-4cc7-a971-a3793f89055e.png)
![image](https://user-images.githubusercontent.com/94054/186694801-27799296-c0b5-4893-81d2-b7e0f18d79a2.png)

Additionally, the fork on GitHub retains the fork metadata instead of being a hard copy (pull and then push manually through the Git CLI).
![image](https://user-images.githubusercontent.com/94054/186692608-d771f3df-4b3e-4f0e-9715-5119070e6d30.png)

Regarding this:
> The main change he made was to replace every occurrence of my name with his name.

I see a mixed situation. I think it demonstrates good intent that the fork author left your name in the "authors" section:
https://github.com/KurtLinder/Silk/blob/66d47aed1fc307d21efa6e15cee7eb27f0658089/Silk/Silk.csproj#L9

Anyways, I could go on and on but hopefully my point is clear: there appears to be some evidence that the fork author was trying to do the right thing when they made the fork.

Finally, I hesitate to "take sides" in this situation. I want to simply characterize how I personally see the situation given the details that are publicly visible. I do appreciate you starting this conversation since I believe we all can learn and benefit from such and open conversation.

## Next steps

- I recommend contacting the fork owner the "Contact owners" link on NuGet.org if you would like to discuss details of why and perhaps work out a more desirable fork situation: https://www.nuget.org/packages/LbsFeldkirch.Silk/2.0.0/ContactOwners. This is driven by you.
- As alluded by @zhhyu above, if you believe the content infringes on an existing trademark or copyright that you hold, you can consider reporting the package for copyright/trademark infringement. Note that this is a very "blunt instrument". I can't advise you on whether this situation meets all of the necessary criteria for a DMCA takedown (my team does not provide legal counsel) but I wanted you to be aware this path does exist.
- NuGet.org team can consider authoring a guide on our documentation site that provides guidelines on how to respectfully fork a package. This would need to be written with great care and would need to seek input from our own internal legal counsel. What are your thoughts on the value of such a guide? Does there exist an agreeable way for this fork author to fork your package? What are the pieces they are missing?

**Comment by joelverhagen** [comment]
Hey @SoftCircuits, we haven't heard back from you so I'll go ahead and close this. Please feel free to reach out to us again with another issue or with our contact page: https://www.nuget.org/policies/Contact.

**Comment by SoftCircuits** [comment]
@joelverhagen Thanks. I did contact the author and it appears he has deprecated his package.
